src/main/java/com/ably/kafka/connect/batch/BatchProcessingThread.java

```diff
@@ -0,0 +1,47 @@
+package com.ably.kafka.connect.batch;
+
+
+import com.ably.kafka.connect.client.AblyClient;
+import com.ably.kafka.connect.client.DefaultAblyBatchClient;
+import com.google.common.collect.Lists;
+import io.ably.lib.types.AblyException;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+/**
+ * A Separate Runnable(Thread) to retrieve the records
+ * from the ConcurrentLinkedQueue and call Ably Batch API
+ * to send records in batch.
+ */
+public class BatchProcessingThread implements Runnable {
+
+    private static final Logger logger = LoggerFactory.getLogger(BatchProcessingThread.class);
+
+    final private ConcurrentLinkedQueue<SinkRecord> records;
+    final private DefaultAblyBatchClient client;
+
+    public BatchProcessingThread(
+            ConcurrentLinkedQueue<SinkRecord> sinkRecords, DefaultAblyBatchClient ablyClient) {
+        this.records = sinkRecords;
+        this.client = ablyClient;
+    }
+
+
+    @Override
+    public void run() {
+        // Process the records.
+        if(this.records.size() > 0) {
+            try {
+                this.client.publishBatch(Lists.newArrayList(this.records.iterator()));
+            } catch (AblyException e) {
+                throw new RuntimeException(e);
```

Comment:
We need to schedule retries somehow, as REST submissions will fail sometimes. We should also anticipate partial successes, where some of the batch needs to be retried. (With idempotent publishing though, it might be easier to just retry the whole batch? @paddybyers ?)

Reply:
I suppose this will be part of error Handling PR.